src/vm.rs

```diff
@@ -225,6 +225,8 @@ pub struct Config {
     /// Allow sh_addr != sh_offset in elf sections. Used in SBFv2 to align
     /// section vaddrs to MM_PROGRAM_START.
     pub enable_elf_vaddr: bool,
+    /// Use the new ELF parser
+    pub new_elf_parser: bool,
```

Comment:
I didn't see this being used anywhere in the code, is it not hooked up yet?

Reply:
It's set to true in `Config::default()`, so tests, cli, fuzzing etc all run with it. Then we'll need a feature for the monorepo. 

The plan with this PR is to enable the new parser unconditionally for all old and SBFv2 files (behind feature flag ofc). Then I would start enforcing restrictions at deploy time *separately* from this work and behind a different flag, or we'll have cases where deploys fail and will be harder to tell whether it's because we're being more strict, or because we have actual parsing bugs.

Reply:
Yeah, I get that, but I must have missed where we toggle between the new and old depending on this config item.

Reply:
That's here https://github.com/solana-labs/rbpf/blob/e0b6e0898ccf12b039f703eedff1ce64d90048bf/src/elf.rs#L433